src/db.ts

```diff
@@ -121,17 +136,44 @@ export class DB {
     }
 
     // Kafka
+    public async queueKafkaMessage(kafkaMessage: ProducerRecord): Promise<void> {
+        this.kafkaMessageQueue.push(kafkaMessage)
+
+        const timeSinceLastFlush = new Date().getTime() - this.lastFlushTime.getTime()
+        if (timeSinceLastFlush > this.flushFrequencyMs || this.kafkaMessageQueue.length >= this.maxQueueSize) {
+            await this.flushKafkaMessages()
+        }
```

Comment:
Why both this and the setInterval?

Reply:
Good question.

I added this to be the main wait condition - this way it's nicely in the path of ingestion and we do block if the queue gets too long.

the interval is the backup condition for making sure we still flush messages even if no messages come in for a longer time.